apps/ios/GuideDogs/Code/Data/Models/Extensions/OSM Entity/GDASpatialDataResultEntity+Typeable.swift

```diff
@@ -12,22 +12,144 @@ extension GDASpatialDataResultEntity: Typeable {
+        guard let category = SuperCategory(rawValue: superCategory) else {
+            print("Failed to map superCategory to SuperCategory enum")
+            return false
+        }
+        print("Mapped category: \(category)")
+        
+        let restaurantTags = [
+            GDLocalizedString("osm.tag.restaurant"),
+            GDLocalizedString("osm.tag.fast_food"),
+            GDLocalizedString("osm.tag.cafe"),
+            GDLocalizedString("osm.tag.bar"),
+            GDLocalizedString("osm.tag.ice_cream"),
+            GDLocalizedString("osm.tag.pub"),
+            GDLocalizedString("osm.tag.coffee_shop")
+            
+        ]
+        
+        let isRestaurantLocation = category == .places && restaurantTags.contains(amenity)
+        
+        print("Place name: \(localizedName)")
+        print("Restaurant location check: \(isRestaurantLocation)")
         
-        return category == .mobility && localizedName.lowercased().contains(GDLocalizedString("osm.tag.bus_stop").lowercased())
+        if isRestaurantLocation {
+            print("Restaurant location found with amenity: \(amenity)")
+        }
+        
+        return isRestaurantLocation
     }
+
+    private func isLandmarks() -> Bool {
+        guard let category = SuperCategory(rawValue: superCategory) else {
+            return false
+        }
+
+        let landmarkTags = [
+            GDLocalizedString("osm.tag.monument"),
+            GDLocalizedString("osm.tag.statue"),
+            GDLocalizedString("osm.tag.museum"),
+            GDLocalizedString("osm.tag.historic"),
+            GDLocalizedString("osm.tag.cathedral")
+        ]
+
+        let lowercasedAmenity = amenity.lowercased()
+
+        let isLandmarkLocation = landmarkTags.contains(lowercasedAmenity)
+
+        // Print the debug statements
+        print("Place name: \(localizedName)")
+        print("Landmark location check: \(isLandmarkLocation)")
+        print("Raw superCategory: \(superCategory)")
+        print("Mapped category: \(category)")
+
+        return isLandmarkLocation 
+    }
+
+    
+    private func isPark() -> Bool {
+        guard let category = SuperCategory(rawValue: superCategory) else {
+            return false
+        }
+
+        // OSM tags related to parks and green spaces
+        let parkTags = [
+            GDLocalizedString("osm.tag.park"),
+            GDLocalizedString("osm.tag.garden"),
+            GDLocalizedString("osm.tag.green_space"),
+            GDLocalizedString("osm.tag.recreation_area"),
+            GDLocalizedString("osm.tag.playground"),
+            GDLocalizedString("osm.tag.nature_reserve"),
+            GDLocalizedString("osm.tag.botanical_garden"),
+            GDLocalizedString("osm.tag.public_garden"),
+            GDLocalizedString("osm.tag.field"),
+            GDLocalizedString("osm.tag.reserve")
+        ]
+
+        let lowercasedAmenity = amenity.lowercased()
+
+        let isParkLocation = parkTags.contains(lowercasedAmenity)
```

Comment:
You'd want to do exact comparison rather than substrings, no? Otherwise you wind up matching e.g. parking lots as parks.